Apps/WebClient/src/ClientApp/src/views/ConfirmTermsOfService.vue

```diff
@@ -0,0 +1,218 @@
+
+    @Ref("termsOfServiceForm")
+    form!: HTMLFormElement;
+
+    private accepted = false;
+    private email = "";
+
+    private userProfileService!: IUserProfileService;
+    private loadingTermsOfService = true;
+
+    private logger!: ILogger;
+
+    private termsOfServiceId = "";
+    private termsOfService = "";
+
+    private get isLoading(): boolean {
+        return this.loadingTermsOfService;
+    }
+
+    private mounted() {
+        this.logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+
+        this.userProfileService = container.get(
+            SERVICE_IDENTIFIER.UserProfileService
+        );
+
+        this.loadTermsOfService();
+    }
+
+    private validations() {
+        return {
+            accepted: { isChecked: sameAs(() => true) },
+        };
+    }
+
+    private loadTermsOfService(): void {
+        this.loadingTermsOfService = true;
+        this.userProfileService
+            .getTermsOfService()
+            .then((result) => {
+                this.logger.debug(
+                    `getTermsOfService result: ${JSON.stringify(result)}`
+                );
+                this.termsOfServiceId = result.id;
+                this.termsOfService = result.content;
+            })
+            .catch((err) => {
+                this.logger.error(err);
+                this.addError({
+                    errorType: ErrorType.Retrieve,
+                    source: ErrorSourceType.TermsOfService,
+                    traceId: undefined,
+                });
+            })
+            .finally(() => {
+                this.loadingTermsOfService = false;
+            });
+    }
+
+    private isValid(param: Validation): boolean | undefined {
+        return param.$dirty ? !param.$invalid : undefined;
+    }
+
+    private onSubmit(event: Event) {
+        this.$v.$touch();
+        if (this.$v.$invalid) {
+            event.preventDefault();
+            return;
+        }
+
+        this.loadingTermsOfService = true;
+
+        this.updateAcceptedTerms({ termsOfServiceId: this.termsOfServiceId })
+            .then(() => {
+                this.redirect();
+            })
+            .catch((err: ResultError) => {
+                this.logger.error(
+                    `Error updating accepted terms of service: ${JSON.stringify(
+                        err
+                    )}`
+                );
+                this.addError({
+                    errorType: ErrorType.Update,
+                    source: ErrorSourceType.Profile,
+                    traceId: err.traceId,
+                });
+            })
+            .finally(() => {
+                this.loadingTermsOfService = false;
+            });
+
+        event.preventDefault();
+    }
+
+    private redirect(): void {
+        this.logger.debug("Redirecting to /profile");
+
+        this.checkRegistration().then(() => {
+            this.$router.push({ path: "/profile" });
+        });
+    }
```

Comment:
Nit: using `checkRegistration` is performing an extra network call to retrieve the user profile when we already have the updated profile from calling `updateAcceptedTerms`.

However, the actions behave differently because the code that sets default values for the user's tutorial preferences is set only in the `checkRegistration` action.

The usage of `checkRegistration` could be removed from here if the preference-setting code from `checkRegistration` was moved into the `setProfileUserData` mutation.

Reply:
Agreed. Not necessary anymore because of backend server side change.